### PR TITLE
Added email parameters for Grafana and ArgoCD

### PR DESCRIPTION
**Pull request by movray**

### CODE REVIEW

File: argocd/argocd/argocd/values.ftl.yaml [hunk 1]
```diff
@@ -102,7 +102,11 @@ argo-cd:
       service.email: |
         host: mailhog.${namePrefix}monitoring.svc.cluster.local
         port: 1025
-        from: argocd-alerts@cloudogu.com
+      <#if argocd.emailFrom?has_content>
```
Comment (schnatterer) [inline argocd/argocd/argocd/values.ftl.yaml:105]: From a functional point of view it's valid to implement the default value here.

In my opinion, this hides the defaults deep in the code.
That's why we implemented all other defaults centrally in `ApplicationConfigurator.DEFAULT_VALUES`.

**Review by schnatterer — CHANGES_REQUESTED** [review]
My suggestion is that you

1. adopt from  
  `config.features['monitoring']['grafana']['emailFrom']` to  
  `config.features['monitoring']['grafanaEmailFrom'] .`  
  This will fix the config file and follow the naming pattern used by the existing `grafanaUrl`.
2. Move the defaults for the 5 parameters to `ApplicationConfigurator.DEFAULT_VALUES` and remove the `<#if x?has_content>` blocks from the templates. This makes them easier to maintain and follows our current approach to centralize all defaults in `ApplicationConfigurator`.

File: src/main/groovy/com/cloudogu/gitops/config/schema/Schema.groovy [hunk 2]
```diff
@@ -83,6 +86,8 @@ class Schema {
          boolean active = true
          String grafanaUrl = ""
          MonitoringHelmSchema helm
+         String grafanaEmailFrom = ""
```
[the remark this answers is not in the document]
Comment (movray) [reply src/main/groovy/com/cloudogu/gitops/config/schema/Schema.groovy:89]: corrected in code

**Review by schnatterer — APPROVED** [review]

**Comment by schnatterer** [comment]
Had to fix apply.sh. 
Then  tested these use cases
* config file, works with dev image (prod has known issue #166) 
* params, works
* defaults, works

Works means the values are written to the expected YAML files in the repositories.
I also watched some actual emails in MailHog.

Only question that remains is: 
**Does Grafana pick up the `grafana-email-to` at all?** I could not find it in Grafana's UI.

Test commands

**config-file:**
``` 
 docker run --rm  -u $(id -u) \
    -v ~/.config/k3d/kubeconfig-gitops-playground.yaml:/home/.kube/config \
  -v $PWD/config.yaml:/config/config.yaml \
    --net=host \                                                                                                                                                                                                  gitops-playground:dev --yes --argocd --base-url=http://localhost  --monitoring -x --config-file=/config/config.yaml --mail
```

with `config.yaml`
```
features: 
  argocd: 
    emailFrom: from@argo.cd
    emailToAdmin: to-admin@argo.cd
    emailToUser: to-user@argo.cd
  monitoring: 
    grafanaEmailFrom: from@grafa.na
    grafanaEmailTo: to@grafa.na
```

**params:**
```
docker run --rm  -u $(id -u) \
   -v ~/.config/k3d/kubeconfig-not-bound.yaml:/home/.kube/config \
    --net=host \
   gitops-playground:dev --yes --argocd --base-url=http://localhost  --monitoring -x --mail --argocd-email-from=from@argo.cd --argocd-email-to-admin=to-admin@argo.cd --argocd-email-to-user=to-user@argo.cd --grafana-email-from=from@grafa.na --grafana-email-to=to@grafa.na
```

**Comment by schnatterer** [comment]
We found out that Grafana not picking up the value that is now `grafana-email-to` is a bug that existed before. A fix is already under way with our upcoming story for external mail server support.

**Pull request merged**